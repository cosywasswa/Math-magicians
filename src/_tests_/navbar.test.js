import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/navbar';

describe('Nav bar component', () => {
    it('should render the navbar', () => {
        const { getByTestId } = render(
            <BrowserRouter>
                <Navbar />
            </BrowserRouter>,
        );
        const bar = getByTestId('navi');
        expect(bar).toBeInTheDocument;
    });

    it('should render the custom link with correct element', () => {
        render(
            <BrowserRouter>
                <Navbar />
            </BrowserRouter>,
        );
        const link = screen.getByTestId('border');
        expect(link).toBeCalled;
    });

    it('should render correctly', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Navbar />
            </BrowserRouter>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
