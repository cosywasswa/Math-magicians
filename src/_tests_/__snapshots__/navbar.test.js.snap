// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav bar component should display correct snapshot 1`] = `
<nav
  className="nav"
  data-testid="navi"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "space-between",
      "margin": "0 1.5rem",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      Math Magicians
    </h1>
  </a>
  <ul
    data-testid="border"
  >
    <li
      className="active border"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "red",
          }
        }
      >
        Home
      </a>
    </li>
    <li
      className=" border"
    >
      <a
        href="/Calculator"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        Calculator
      </a>
    </li>
    <li
      className=" lastLink"
    >
      <a
        href="/Quotes"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Nav bar component should render correct snapshot 1`] = `
<nav
  className="nav"
  data-testid="navi"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "space-between",
      "margin": "0 1.5rem",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      Math Magicians
    </h1>
  </a>
  <ul
    data-testid="border"
  >
    <li
      className="active border"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "red",
          }
        }
      >
        Home
      </a>
    </li>
    <li
      className=" border"
    >
      <a
        href="/Calculator"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        Calculator
      </a>
    </li>
    <li
      className=" lastLink"
    >
      <a
        href="/Quotes"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Nav bar component should render correctly 1`] = `
<nav
  className="nav"
  data-testid="navi"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "space-between",
      "margin": "0 1.5rem",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      Math Magicians
    </h1>
  </a>
  <ul
    data-testid="border"
  >
    <li
      className="active border"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "red",
          }
        }
      >
        Home
      </a>
    </li>
    <li
      className=" border"
    >
      <a
        href="/Calculator"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        Calculator
      </a>
    </li>
    <li
      className=" lastLink"
    >
      <a
        href="/Quotes"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
